pkg/scheduler/cache/refermetrics/metrics_client.go

```diff
@@ -0,0 +1,34 @@
+/*
+ Copyright 2023 The Volcano Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package refermetrics
+
+import (
+	"context"
+	"errors"
+)
+
+type MetricsClient interface {
+	NodeMetricsAvg(ctx context.Context, nodeName string, period string) (*NodeMetrics, error)
+}
+
+func NewMetricsClient(metricsConf map[string]string) (MetricsClient, error) {
+	address := metricsConf["address"]
+	if len(address) == 0 {
+		return nil, errors.New("metrics address is empty")
+	}
+	return &PrometheusMetricsClient{address: address}, nil
```

Comment:
How to support other metrics system like ELK, is there a whole picture design?

Reply:
why return a `PrometheusMetricsClient` in a generic `NewMetricsClient` func. It's better use a design pattern to support multiple client.

Reply:
@william-wang will add other type of `NewMetricsClient` in next PR.